### PR TITLE
new(driver): add 2 new scap stats 

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area API-version

/area driver-kmod

/area driver-bpf

/area driver-modern-bpf

/area libscap-engine-bpf

/area libscap-engine-kmod

/area libscap-engine-modern-bpf

/area libscap

/area libpman

/area libsinsp

**Does this PR require a change in the driver versions?**

/version driver-API-version-major

**What this PR does / why we need it**:

This PR adds 2 new kernel stats:

- n_drops_buffer_close_exit: CLOSE exit event drops.
- n_drops_buffer_proc_exit: SCHED_PROC_EXIT event drops.

They could be very useful when debugging possible mem-leak or high memory usage since the `PROC_EXIT` event is strictly related to the number of threads in our table while the `CLOSE_X` is related to the number of file descriptors.

For the same debugging purpose I added some logs when the capture is closed to print the dimension of the thread table and the total number of fd still opened.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

This PR requires an API major bump since the stat struct sent by the drivers is changed

**Does this PR introduce a user-facing change?**:

```release-note
new(driver): add 2 new scap stats
```

### CODE REVIEW

File: userspace/libsinsp/sinsp.cpp [hunk 1]
```diff
@@ -1967,6 +1955,55 @@ void sinsp::get_capture_stats(scap_stats* stats) const
...
+	get_capture_stats(&stats);
+
+	g_logger.format(sev,
+		"\nn_evts:%" PRIu64
+		"\nn_drops:%" PRIu64
+		"\nn_drops_buffer:%" PRIu64
+		"\nn_drops_buffer_clone_fork_enter:%" PRIu64
+		"\nn_drops_buffer_clone_fork_exit:%" PRIu64
+		"\nn_drops_buffer_execve_enter:%" PRIu64
+		"\nn_drops_buffer_execve_exit:%" PRIu64
```
Comment (Andreagit97) [inline userspace/libsinsp/sinsp.cpp:1970]: I've slightly changed the way in which we print the log. Now we don't have anymore a single line but we put a `\n` after each stat. This is an output example

```
08-21 11:32:31.914754 
n_evts:211937
n_drops:0
n_drops_buffer:0
n_drops_buffer_clone_fork_enter:0
n_drops_buffer_clone_fork_exit:0
n_drops_buffer_execve_enter:0
n_drops_buffer_execve_exit:0
n_drops_buffer_connect_enter:0
n_drops_buffer_connect_exit:0
n_drops_buffer_open_enter:0
n_drops_buffer_open_exit:0
n_drops_buffer_dir_file_enter:0
n_drops_buffer_dir_file_exit:0
n_drops_buffer_other_interest_enter:0
n_drops_buffer_other_interest_exit:0
n_drops_buffer_close_exit:0
n_drops_buffer_proc_exit:0
n_drops_scratch_map:0
n_drops_pf:0
n_drops_bug:0

08-21 11:32:31.914981 total threads in the table:1522, total fds in all threads:5319
```

So as you can see the output is quite small but if we should face issues in the future we could always revert it on one line, WDYT?

Comment (incertum) [reply userspace/libsinsp/sinsp.cpp:1970]: It's used for debugging aka no problems at all would say

File: driver/bpf/fillers.h [hunk 2]
```diff
@@ -190,6 +190,12 @@ FILLER_RAW(terminate_filler)
 					++state->n_drops_buffer_other_interest_enter;
 				}
 				break;
+			case PPME_PROCEXIT_E:
```
Comment (incertum) [inline driver/bpf/fillers.h:193]: @Andreagit97 while we are here, I think it was during the loginuid PR where I saw we could cleanup and remove some of the old `PPME_SYSCALL_CLONE_*` events. Could we sneak this cleanup in?

Also do you think the categories `n_drops_buffer_dir_file*` and `n_drops_buffer_other_interest*` could benefit from some updates while here?

**Review by incertum — COMMENTED** [review]
@Andreagit97 great idea, love it and you know I love any useful stats always 😄 

On Falco side the new metrics will pick up these new stats automatically, but the old drop internal messages / stats would need an update PR as well.

Just left a small ask for some unrelated cleanup to consider while you are here. Can then approve tomorrow!

File: userspace/libscap/examples/01-open/README.md [hunk 3]
```diff
@@ -111,6 +111,8 @@ Number of dropped events caused by full buffer (n_drops_buffer_dir_file_enter sy
 Number of dropped events caused by full buffer (n_drops_buffer_dir_file_exit syscall category): 0
 Number of dropped events caused by full buffer (n_drops_buffer_other_interest_enter syscall category): 0
 Number of dropped events caused by full buffer (n_drops_buffer_other_interest_exit syscall category): 0
+Number of dropped events caused by full buffer (n_drops_buffer_close_exit syscall category): 0
```
Comment (incertum) [inline userspace/libscap/examples/01-open/README.md:114]: oh @Andreagit97 we need to update the README as we now have the new scap stats v2 output in the scap-open example binary. That's on my I forgot it back then ...

File: driver/bpf/fillers.h (hunk 2 above)
Comment (Andreagit97) [reply driver/bpf/fillers.h]: Done :) 

> Also do you think the categories n_drops_buffer_dir_file* and n_drops_buffer_other_interest* could benefit from some updates while here?

During my experiments I've never used other stats so at the moment I would say no :thinking:

File: userspace/libscap/examples/01-open/README.md (hunk 3 above)
Comment (Andreagit97) [reply userspace/libscap/examples/01-open/README.md]: done :)

File: driver/bpf/fillers.h
Comment (FedeDP) [inline driver/bpf/fillers.h:1]: Nice cleanup :D

**Review by incertum — APPROVED** [review]
/approve

**Pull request merged**